mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/HighLevelMqttSource.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.impl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.alpakka.mqtt.streaming.scaladsl._
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, RestartFlow, Source, SourceQueueWithComplete}
+import akka.stream.{Materializer, OverflowStrategy}
+import akka.{Done, NotUsed}
+
+import scala.collection.immutable
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Internal API
+ */
+@InternalApi
+private[streaming] object HighLevelMqttSource {
+
+  def atMostOnce(
+      mqttClientSession: MqttClientSession,
+      transportSettings: MqttTransportSettings,
+      restartSettings: MqttRestartSettings,
+      connectionSettings: MqttConnectionSettings,
+      subscriptions: MqttSubscribe
+  ): Source[Publish, Future[immutable.Seq[(String, ControlPacketFlags)]]] = {
+    type Out = Publish
+
+    val sendAcknowledge: SourceQueueWithComplete[Command[Nothing]] => PartialFunction[Event[Nothing], Out] =
+      commands => {
+        case Event(publish @ Publish(_, _, Some(packetId), _), _) =>
+          commands.offer(Command(PubAck(packetId)))
+          publish
+        case Event(publish: Publish, _) =>
+          publish
+      }
+
+    createOnMaterialization[Out](mqttClientSession,
+                                 transportSettings,
+                                 restartSettings,
+                                 connectionSettings,
+                                 subscriptions,
+                                 sendAcknowledge)
+  }
+
+  def atLeastOnce[Out](
```

Comment:
I'm unsure what the conventions should be, but we've been using type params as `A`, `B`, `C` etc. Perhaps we should be consistent even if it means changing the other code. WDYT?

Reply:
I can rename it. For type-params which have an intention I often use "real" names.

Reply:
Your call of course. I certainly don't mind `Out` - in fact, I quite like it... just wondering if there should be consistency or even whether it matters.

Reply:
I like your ambition when working with Alpakka, though we have to accept we can't reach consistency on that level, I'm afraid.

Reply:
I'm a very ambitious person. ;-)